tests/rptest/tests/tx_admin_api_test.py

```diff
@@ -0,0 +1,233 @@
+            target_node_id = next(
+                filter(lambda r: r["id"] != partition["leader"],
+                       partition["replicas"]))["id"]
+            self.logger.debug(
+                f"Transfering leader from {partition['leader']} to {target_node_id}"
+            )
+
+            # build the transfer url
+            meta = kc.metadata()
+            brokers = meta["brokers"]
+            source_broker = next(
+                filter(lambda b: b["id"] == partition["leader"], brokers))
+
+            target_broker = None
+
+            if new_leader != None:
+                target_broker = new_leader
+            else:
+                target_broker = next(
+                    filter(lambda b: b["id"] == target_node_id, brokers))
+            self.logger.debug(f"Source broker {source_broker}")
+            self.logger.debug(f"Target broker {target_broker}")
+
+            # Send the request to any host, they should redirect to
+            # the leader of the partition.
+            partition_id = partition['partition']
+
+            admin = Admin(self.redpanda)
+            admin.partition_transfer_leadership("kafka", self.topic,
+                                                partition_id, target_node_id)
+
+            def transfer_complete():
+                for _ in range(3):  # just give it a moment
+                    time.sleep(1)
+                    meta = kc.metadata()
+                    partition = next(
+                        filter(lambda p: p["partition"] == partition_id,
+                               meta["topics"][0]["partitions"]))
+                    if partition["leader"] == target_node_id:
+                        return True
+                return False
+
+            wait_until(lambda: transfer_complete(),
+                       timeout_sec=30,
+                       backoff_sec=5,
+                       err_msg="Transfer did not complete")
+
+    def _get_partitions(self, kc, topic_name):
+        partitions = list()
+
+        def get_partitions(topic_name):
+            meta = kc.metadata()
+            topics = meta["topics"]
+            topic_info = list(
+                filter(lambda r: r["topic"] == topic_name, topics))
+            assert (len(topic_info) == 1)
+
+            for partition in topic_info[0]['partitions']:
+                if partition["leader"] <= 0:
+                    return False
+
+            for partition in topic_info[0]['partitions']:
+                partitions.append(partition)
+
+            return True
+
+        wait_until(lambda: get_partitions(topic_name),
+                   timeout_sec=30,
+                   backoff_sec=2,
+                   err_msg="No partition with leader available")
+
+        return partitions
+
+    def extract_pid(self, tx):
+        return (tx["producer_id"]["id"], tx["producer_id"]["epoch"])
+
+    @cluster(num_nodes=3)
+    def test_simple_get_transaction(self):
+        self.producer1 = ck.Producer({
```

Comment:
we don't use `producer1` & `producer2` outside of this method, let's keep them local instead of class members; it limits the scope and hints a reader that they may not to think about them anymore